### PR TITLE
MES-6282 / Waiting Room To Car

### PR DESCRIPTION
**Pull request by matthew2564**
## Description
- Implement WRTC page for Cat B
- Remove a lot of duplicated code into basepage for re-use
- Port over all common WRTC components

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
Screenshot 2021-05-07 at 15 33 44
Screenshot 2021-05-07 at 14 56 37
Screenshot 2021-05-07 at 14 50 38
Screenshot 2021-05-07 at 13 44 06
Screenshot 2021-05-07 at 13 44 01

### CODE REVIEW

File: src/app/pages/waiting-room-to-car/components/accompaniment/accompaniment.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div [formGroup]="formGroup">
+    <input [id]="formControlName" type="checkbox" class="gds-checkbox"
+            [formControlName]="formControlName"
+            (change)="accompanimentChanged()"
+            [value]="accompanimentType">
+    <label  [for]="formControlName">{{accompanimentType}}</label>
+</div>
```
Comment (Cullima) [inline src/app/pages/waiting-room-to-car/components/accompaniment/accompaniment.html]: can you add end line in?

File: src/global.scss [hunk 2]
```diff
@@ -54,3 +54,8 @@
     text-align: center;
   }
 }
+
+.ion-padding-left {
```
Comment (Cullima) [inline src/global.scss]: why do we need to do this?

**Review by Cullima — APPROVED** [review]

File: src/app/pages/waiting-room-to-car/components/accompaniment/accompaniment.html (hunk 1 above)
Comment (matthew2564) [reply src/app/pages/waiting-room-to-car/components/accompaniment/accompaniment.html]: will do

File: src/global.scss (hunk 2 above)
Comment (matthew2564) [reply src/global.scss]: padding-left attribute is deprecated so created css class to replicate functionality

File: src/app/pages/waiting-room-to-car/cat-b/components/waiting-room-to-car.cat-b.components.module.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { CommonModule } from '@angular/common';
+import { IonicModule } from '@ionic/angular';
+import { ReactiveFormsModule } from '@angular/forms';
+
+import { ComponentsModule } from '@components/common/common-components.module';
+import { TellMeQuestionCardComponent } from './tell-me-question-card/tell-me-question-card';
+import { TellMeQuestionComponent } from './tell-me-question/tell-me-question';
+import { TellMeQuestionOutcomeComponent } from './tell-me-question-outcome/tell-me-question-outcome';
+import { InstructorRegistrationComponent } from './instructor-registration/instructor-registration';
+import { DirectivesModule } from '../../../../../directives/directives.module';
```
Comment (andrewsetterfield) [inline src/app/pages/waiting-room-to-car/cat-b/components/waiting-room-to-car.cat-b.components.module.ts:11]: this looks so wrong now. add Alias?

**Review by tomcrawleyy — APPROVED** [review]

File: src/app/pages/waiting-room-to-car/cat-b/components/waiting-room-to-car.cat-b.components.module.ts (hunk 3 above)
Comment (matthew2564) [reply src/app/pages/waiting-room-to-car/cat-b/components/waiting-room-to-car.cat-b.components.module.ts:11]: @directives doesnt exist in dev so is valid as per this merge

File: src/app/pages/waiting-room-to-car/components/vehicle-registration/vehicle-registration.scss [hunk 4]
```diff
@@ -0,0 +1,6 @@
+:host {
+  ion-input {
+    --ion-background-color: red;
```
Comment (andrewsetterfield) [inline src/app/pages/waiting-room-to-car/components/vehicle-registration/vehicle-registration.scss]: var(--gds-red?)

**Review by andrewsetterfield — APPROVED** [review]

File: src/app/pages/waiting-room-to-car/components/vehicle-registration/vehicle-registration.scss (hunk 4 above)
Comment (matthew2564) [reply src/app/pages/waiting-room-to-car/components/vehicle-registration/vehicle-registration.scss]: will do

**Pull request merged**